internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -576,6 +703,107 @@ func (r *AgentServiceConfigReconciler) newAssistedCM(ctx context.Context, log lo
 	return cm, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) newImageServiceDeployment(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) (client.Object, controllerutil.MutateFn, error) {
+	deploymentLabels := map[string]string{
+		"app": imageServiceName,
+	}
+
+	container := corev1.Container{
+		Name:            imageServiceName,
+		Image:           ImageServiceImage(),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: int32(imageHandlerPort.IntValue()),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: []corev1.EnvVar{
+			{Name: "LISTEN_PORT", Value: imageHandlerPort.String()},
+			{Name: "RHCOS_VERSIONS", Value: r.getOSImages(log, instance)},
+			{Name: "HTTPS_CERT_FILE", Value: "/etc/tls/tls.crt"},
+			{Name: "HTTPS_KEY_FILE", Value: "/etc/tls/tls.key"},
+			{Name: "HTTPS_CA_FILE", Value: "/etc/cabundle/service-ca.crt"},
```

Comment:
Looks like https://github.com/openshift/assisted-image-service/pull/19 is required for this, right?

Reply:
Yes.

Reply:
This path doesn't actually matter, right? Could we make it something that actually looks less special?
Looking at this the first time I thought this path was something that got handled automatically rather than just somewhere to put a file that we read the content from. Was also worried that we might be overwriting something else important to the system.

Same applies for the cert and key `/etc/tls` looks a bit too official.